modules/library/metadata/src/main/java/org/geotools/factory/FactoryCreator.java

```diff
@@ -134,15 +136,27 @@ public FactoryCreator(final Collection<Class<?>> categories) {
      * @throws FactoryRegistryException if the factory can't be created for some other reason.
      */
     @Override
+    @Deprecated // TODO document
```

Comment:
I guess you are deprecating this to move towards Predicate use rather than javax.imageio.spi.ServiceRegistry.Filter.

I would not mind checking with the GeoTools PMC about making this a breaking change (forcing the use of getFactory( category, filter, hints, key).

Reply:
Sure, let's discuss that at the call.